subscription/src/search.py

```diff
@@ -0,0 +1,186 @@
+            port = env_vars.get_env_var_from_parameter_store(parameter_name=port_param_name)
+            host = env_vars.get_env_var_from_parameter_store(parameter_name=host_param_name)
+            context = env_vars.get_env_var_from_parameter_store(parameter_name=context_param_name)
+
+            # The context already contains the forward / so we don't need it here.
+            self.public_search_url = f"{protocol}://{host}:{port}{context}"
+            logger.debug(f"Subscription Worker Public Search URL: {self.public_search_url}")
+
+    def get_public_search_url(self):
+        """This function returns the public search URL if it has already been constructed, otherwise it constructs the URL and then returns it."""
+        if not self.public_search_url:
+            self.get_public_search_url_from_parameter_store()
+        return self.public_search_url
+
+    def get_token_from_parameter_store(self):
+        """This function returns the token for the search service. Otherwise the 
+        environment name that is used for the parameter store prefix is obtained from an environment variable. This variable is used to 
+        get the parameter store ingest values to construct the access control service URL."""
+
+        # token is for local development
+        token = os.getenv("CMR_ECHO_SYSTEM_TOKEN")
+
+        if token:
+            self.token = token
+            return
+        else:
+            # This block gets the token from the AWS parameter store.
+            environment_name = os.getenv("ENVIRONMENT_NAME")
+
+            if not environment_name:
+                logger.error("ENVIRONMENT_NAME environment variable is not set")
+                raise ValueError("ENVIRONMENT_NAME environment variable is not set")
+
+            # construct the the token parameter names from the environment variable
+            env_name = environment_name.lower()
+            token_name = f"/{env_name}/ingest/CMR_ECHO_SYSTEM_TOKEN"
+
+            env_vars = Env_Vars()
+            self.token = env_vars.get_env_var_from_parameter_store(token_name, True)
+
+    def get_token(self):
+        """This function returns the token if it has already been constructed, otherwise it gets the token and then returns it."""
+        if not self.token:
+            self.get_token_from_parameter_store()
+        return self.token
+
+    def get_concept(self, concept_id, revision_id):
+        """This function calls search using a token, and a CMR concept id to get
+            a granule by concept id in umm_json format."""
+
+        # Set the search concepts URL.
+        if revision_id:
+            url = f"{self.get_public_search_url()}/concepts/{concept_id}/{revision_id}.umm_json"
+        else:
+            url = f"{self.get_public_search_url()}/concepts/{concept_id}.umm_json"
+
+        # Set the headers
+        headers = {
+            "Authorization": self.get_token(),
+            "Client-Id": "subscription-worker"
```

Comment:
while your at it, I would set the User-Agent to the same thing. They show up in logs differently and it is best to not let requests report what version it is which helps no one.

Reply:
I have added it.